src/scripts/import-needs-assessment-data/types.d.ts

```diff
@@ -0,0 +1,56 @@
+export enum UploadWorkflowStatus {
+  PROCESSING = "PROCESSING",
+  ORIGINAL_DATA_INVALID = "ORIGINAL_DATA_INVALID",
+  DUPLICATE_CHECK_ERROR = "DUPLICATE_CHECK_ERROR",
+  ALREADY_EXISTS = "ALREADY_EXISTS",
+  UPLOAD_ERROR = "UPLOAD_ERROR",
+  SUCCESS = "SUCCESS",
+  OTHER = "OTHER",
+}
+
+export type UploadWorkflow<T> = {
+  data: T;
+  orig: string | T; // Union type allowing either a csv string or parsed JSON object
+  status: UploadWorkflowStatus;
+  logs: string[];
+};
+
+export type UploadWorkflowResults<T> = {
+  [Property in keyof UploadWorkflowStatus]: T[];
+};
+
+export type RegionUploadWorkflow = UploadWorkflow<Region>;
+export type RegionUploadWorkflowResults = UploadWorkflowResults<Region>;
+export type SubregionUploadWorkflow = UploadWorkflow<Subregion>;
+export type SubregionUploadWorkflowResults = UploadWorkflowResults<Subregion>;
+
+export type Region = {
+  region?: string;
+};
+
+export type Subregion = {
+  subregion?: string;
+};
+
+export type NeedAssessment = {
+  id?: string;
```

Comment:
Are ids actually optional or should this be `id: string`? And so on, like Region etc.

Reply:
These are likely artifects from the original landing site to make types work while processing the data there.

In the actual needs JSON we're importing:

- every need has an `id`
- every need has a `survey` with all sub-fields
- every need has a `place` and a `place.region`, but some needs don't have a `place.subregion`
- every need has a `product`, including `product.category`,  `product.item`, and `product.unit`.  some needs don't have a `product.ageGender` or `product.sizeStyle` as those are optional attributes.
- a small # of needs don't have a `need.need` (aka the amount of items needed).  maybe just drop those as an error case?

I believe these types model the import data, and might not actually be the shape of the data as it's created in Strapi.  For example, the `need.id` is the need id from Gatsby which won't be what's in Strapi (tho there will still be an ID from Strapi there).

@veryeli how do you suggest @ll-zerr proceeds here?

Reply:
Got this fixed up!